### PR TITLE
uses e2etests for scaling tests

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/e2etests/pull/2.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -65,6 +65,10 @@ required = [
   branch = "master"
   name = "github.com/giantswarm/e2etemplates"
 
+[[constraint]]
+  branch = "scaling"
+  name = "github.com/giantswarm/e2etests"
```
Comment (xh3b4sd) [inline Gopkg.toml]: Waiting for https://github.com/giantswarm/e2etests/pull/2 to be merged.

File: integration/test/scaling/main_test.go [hunk 2]
```diff
@@ -48,7 +55,39 @@ func TestMain(m *testing.M) {
...
+	}
+
 	c = aws.NewClient()
+}
 
+// TestMain allows us to have common setup and teardown steps that are run
+// once for all the tests https://golang.org/pkg/testing/#hdr-Main.
+func TestMain(m *testing.M) {
 	setup.WrapTestMain(c, g, h, m)
 }
```
Comment (xh3b4sd) [inline integration/test/scaling/main_test.go]: This is the wrapping now. Super clean.

File: integration/test/scaling/main_test.go [hunk 3]
```diff
@@ -7,30 +7,37 @@ import (
...
 	g *framework.Guest
 	h *framework.Host
-	c *aws.Client
+	s *scaling.Scaling
 )
 
-// TestMain allows us to have common setup and teardown steps that are run
-// once for all the tests https://golang.org/pkg/testing/#hdr-Main.
-func TestMain(m *testing.M) {
+func init() {
```
Comment (xh3b4sd) [inline integration/test/scaling/main_test.go]: Moved the initialization to `init`. Seems to be a good fit here.

File: integration/test/scaling/scaling_test.go [hunk 4]
```diff
@@ -3,91 +3,13 @@
...
-}
-
-func numberOfWorkers(clusterName string) (int, error) {
-	cluster, err := h.AWSCluster(clusterName)
+func Test_Scaling_Workers(t *testing.T) {
+	err := s.Test(context.Background())
 	if err != nil {
-		return 0, microerror.Mask(err)
+		t.Fatalf("%#v", err)
 	}
```
Comment (xh3b4sd) [inline integration/test/scaling/scaling_test.go]: This is the test now. 🎉

Comment (indyfree) [reply integration/test/scaling/scaling_test.go]: 🎉

**Comment by xh3b4sd** [comment]
@fgimenez @rossf7 I see these issues now in the e2e tests. What can I do about this? 

```
2018/06/13 15:04:12 {"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/resource_wrapper.go:74","event":"update","level":"warning","message":"retrying due to error","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/ci-cur-45b63-5e958","resource":"hostedzonev12","stack":"[{/go/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/resource_wrapper.go:67: } {/go/src/github.com/giantswarm/aws-operator/service/controller/v12/resource/hostedzone/resource.go:53: } {/go/src/github.com/giantswarm/aws-operator/service/controller/v12/resource/hostedzone/resource.go:149: zone = \"gauss.eu-central-1.aws.gigantic.io\"} {hosted zone not found}]","time":"2018-06-13T14:33:26.28572+00:00","underlyingResource":"hostedzonev12"}
```

```
2018/06/13 15:21:14 {"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/crud_resource_ops_wrapper.go:71","event":"delete","function":"GetCurrentState","level":"warning","message":"retrying due to error","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/ci-wip-45b63-5e958","resource":"kmskeyv13","stack":"[{/go/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/crud_resource_ops_wrapper.go:64: } {/go/src/github.com/giantswarm/aws-operator/service/controller/v13/resource/encryptionkey/current.go:19: } {/go/src/github.com/giantswarm/aws-operator/service/controller/v13/encrypter/kms/kms.go:130: } {/go/src/github.com/giantswarm/aws-operator/service/controller/v13/encrypter/kms/kms.go:206: } {NotFoundException: Alias arn:aws:kms:eu-central-1:270935918670:alias/ci-wip-45b63-5e958 is not found.\n\tstatus code: 400, request id: 69299e76-6f1d-11e8-8bbe-31a93f20fb76}]","time":"2018-06-13T15:21:14.479649+00:00","underlyingResource":"kmskeyv13"}
```

**Comment by fgimenez** [comment]
About the second one, looks like the masked aws api is not being properly detected, this #1015 should fix it.

**Comment by xh3b4sd** [comment]
@rossf7 the first issue, isn't this something you worked on IIRC? Any idea what I can do about it? Or should it already be fixed? 

```
zone = \"gauss.eu-central-1.aws.gigantic.io\"} {hosted zone not found}
```

**Comment by rossf7** [comment]
@xh3b4sd For the hosted zone error is this in `v12`?

I disabled it there because it doesn't work with custom DNS in `viking`. Have you retried after a rebase?

**Comment by xh3b4sd** [comment]
Ack. They are running again. Lets see. Thanks for the heads up.

**Comment by xh3b4sd** [comment]
Scaling tests work now. Going.

**Pull request merged**